debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/TypeRegistry.java

```diff
@@ -405,6 +405,9 @@ private PostgresType loadType(Connection connection, PreparedStatement statement
             while (rs.next()) {
                 PostgresType result = createTypeBuilderFromResultSet(connection, rs, typeInfo, sqlTypeMapper).build();
                 addType(result);
+                if (!connection.getAutoCommit() ) {
```

Comment:
Why does it commit here, in particular for each iteration in the result set?

Reply:
@gunnarmorling Please not the `return` statement just after the commit, it terminates the loop...